library/src/scala/runtime/LazyVals.scala

```diff
@@ -37,6 +38,41 @@ object LazyVals {
 
   /* ------------- Start of public API ------------- */
 
+  /**
+   * Used to indicate the state of a lazy val that is being
+   * evaluated and of which other threads await the result.
+   */
+  final class Waiting:
```

Comment:
Have you considered extending `java.util.concurrent.CountDownLatch(1)`, and directly using `countDown()` (resp. `await()`) instead of `release()` (resp. `awaitRelease()`)? Supposedly, each JVM/JDK may have an implementation of `CountDownLatch` that is optimal for its optimization patterns, and so that may be faster than a custom implementation?

Reply:
Thanks, that's a good idea.